client/src/components/Bills/BillList.js

```diff
@@ -0,0 +1,144 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import BillForm from "./BillForm";
+
+const emptyBill = {
+  transaction_date: "",
+  subtotal: null,
+  tax_rate: null,
+  tax_amount: null,
+  tip_rate: null,
+  tip_total: null,
+  who_paid: null,
+  paid_up: null,
+  bill_notes: "",
+  full_total: null,
+};
+
+function BillList() {
+  //This needs an empty array, or the whole thing breaks
+  const [bills, setBills] = useState([]);
+  const [editBillById, setEditBillById] = useState(null);
+
+  useEffect(() => {
+    fetch("http://localhost:5005/db/bills")
+      .then((response) => response.json())
+      .then((billData) => {
+        setBills(billData);
+      });
+  }, []);
+
+  //  const deleteP Contacts
+  const deleteBill = async (bill_id) => {
+    try {
+      const deleteResponse = await fetch(
+        `http://localhost:5005/db/bills/${bill_id}`,
+        {
+          method: "DELETE",
+        }
+      );
+      if (deleteResponse.status === 200) {
+        setBills(bills.filter((bill) => bill.bill_id !== bill_id));
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  //PUT stuff
+  // grabs the id of the post to be editted
+  const editBill = (bill) => {
+    const editId = bill.bill_id;
+    console.log(editId);
+    setEditBillById(editId);
+  };
+
+  const updateBill = async (updatedBillInfo) => {
+    setBills((bills) => {
+      const newListBills = [];
```

Comment:
💡 there's a more efficient way of doing this, but I don't think it adds anything to the app.